### PR TITLE
Indexed api. fixes #25

### LINKED ISSUE #25: Support basic CRUD operations on /data service
Let's support the basics of a data API:

```
GET /data/api/{key}
PUT /data/api/{key}
POST /data/api/{key}
DELETE /data/api/{key}
```

To start, we should just allow things that can be serialized to JSON.

### PR DESCRIPTION
**Pull request by dsych**
This is a basic REST style API into indexedDB done through [Google Workbox](https://developers.google.com/web/tools/workbox/) (caching) and [Dexie.js](https://github.com/dfahlander/Dexie.js) (a wrapper for IndexedDB).
The way it works:
1. Workbox registers a bunch of routes to be precached (full offline support) and a REST style api on the "/data/api" route.
2. Whenever there is a request that goes to "/data/api", workbox intercepts it and makes an indexedDb query through dexie api.

A couple of things to note:
* There is a temp UI for ease of interactions with the endpoint.
* Currently supported methods: GET, POST, PUT, DELETE.
* For the get, syntax of the url is supposed to be "tableName/propertyToBeRetrieved/valueToBeMatched"
* For PUT and POST, the request body is supposed to be a valid JSON, which means putting quotes(") around variable name e.g. {"a": 1, "b":"someString"}.
* Additional information could be found on the UI itself. 
* `GH_PAGES` variable needs to be present in travis build that points to desired gh-pages url. E.g. `https://humphd.github.io/next/`

Fixing #25 
Deployed version could be found [here](https://dsych.github.io/next/)

### CODE REVIEW

**Comment by Fatehsandhu** [comment]
@dsych We will have to wait on this until my PR is merged as well.

**Comment by dsych** [comment]
@Fatehsandhu I don't see any other PR. Are there conflicts in tech?

**Comment by Fatehsandhu** [comment]
@dsych No, no conflicts. Mine is going to be a PR for Workbox integration. I will have it up soon.
Thanks

**Comment by dsych** [comment]
Update:
This code is **no longer** attempts to run offline, since it not needed for dev only resources e.g. UI into the DB.

File: src/db/db-worker.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            new workbox.expiration.Plugin({
+                maxEntries: 50,
+                maxAgeSeconds: 30 * 24 * 60 * 60, // 30 Days
+            }),
+        ],
+    })
+);
+
+workbox.routing.registerRoute(
+    /\b(docs|assets|bin|blog|data\/(reset|download|upload)|lib|www).*.html/,
```
Comment (humphd) [inline src/db/db-worker.js]: Let's remove `.html` off the end.

File: src/db/db-worker.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+    workbox.strategies.cacheFirst({
+        cacheName: 'htmls-cache',
```
Comment (humphd) [inline src/db/db-worker.js]: Let's change this to `content-cache`

File: src/db/db-worker.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+// @ts-ignore
+workbox.setConfig();
+
+workbox.routing.registerRoute(
+    /\b(io\/from\/dataurl|io\/in|io\/share|io\/archive|terminal|io\/reset).*/,
+    workbox.strategies.cacheFirst({
+        cacheName: 'io-cache',
+        plugins: [
+            new workbox.expiration.Plugin({
+                maxEntries: 50,
```
Comment (humphd) [inline src/db/db-worker.js]: Let's see if we need this, and if it can be larger.

File: src/db/db-worker.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    })
+);
+
+workbox.routing.registerRoute(
+    /\b(docs|assets|bin|blog|data\/(reset|download|upload)|lib|www).*.html/,
+    workbox.strategies.cacheFirst({
+        cacheName: 'htmls-cache',
+        plugins: [
+            new workbox.expiration.Plugin({
+                maxEntries: 50,
```
Comment (humphd) [inline src/db/db-worker.js]: Same thing here, let's research this and do better if possible

**Comment by humphd** [comment]
This is good to go with the extra commit that is coming from @dsych.  He's good to merge when he's done.

**Pull request merged**